# Import Libraries

In [1]:
import pandas as pd
import numpy as np

# Load Dataset

In [4]:
df = pd.read_csv("Dataset/Clean_Dataset.csv")

df.head()

,Unnamed: 0,airline,flight,source_city,departure_time,stops,arrival_time,destination_city,class,duration,days_left,price
0,0,SpiceJet,SG-8709,Delhi,Evening,zero,Night,Mumbai,Economy,2.17,1,5953
1,1,SpiceJet,SG-8157,Delhi,Early_Morning,zero,Morning,Mumbai,Economy,2.33,1,5953
2,2,AirAsia,I5-764,Delhi,Early_Morning,zero,Early_Morning,Mumbai,Economy,2.17,1,5956
3,3,Vistara,UK-995,Delhi,Morning,zero,Afternoon,Mumbai,Economy,2.25,1,5955
4,4,Vistara,UK-963,Delhi,Morning,zero,Morning,Mumbai,Economy,2.33,1,5955


## Dataset Overview

In [5]:
print("="*50)
print("Dataset Shape :", df.shape)

print("\nRows :", df.shape[0])
print("Columns :", df.shape[1])

print("\nColumn Names")
print(df.columns.tolist())

print("\nData Types")
print(df.dtypes)

Dataset Shape : (300153, 12)

Rows : 300153
Columns : 12

Column Names
['Unnamed: 0', 'airline', 'flight', 'source_city', 'departure_time', 'stops', 'arrival_time', 'destination_city', 'class', 'duration', 'days_left', 'price']

Data Types
Unnamed: 0            int64
airline              object
flight               object
source_city          object
departure_time       object
stops                object
arrival_time         object
destination_city     object
class                object
duration            float64
days_left             int64
price                 int64
dtype: object


##  Dataset and Statistical Summary

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300153 entries, 0 to 300152
Data columns (total 12 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Unnamed: 0        300153 non-null  int64  
 1   airline           300153 non-null  object 
 2   flight            300153 non-null  object 
 3   source_city       300153 non-null  object 
 4   departure_time    300153 non-null  object 
 5   stops             300153 non-null  object 
 6   arrival_time      300153 non-null  object 
 7   destination_city  300153 non-null  object 
 8   class             300153 non-null  object 
 9   duration          300153 non-null  float64
 10  days_left         300153 non-null  int64  
 11  price             300153 non-null  int64  
dtypes: float64(1), int64(3), object(8)
memory usage: 27.5+ MB


In [11]:
df.describe(include="all").T

,count,unique,top,freq,mean,std,min,25%,50%,75%,max
Unnamed: 0,300153.0,NaN,NaN,NaN,150076.0,86646.852011,0.0,75038.0,150076.0,225114.0,300152.0
airline,300153,6,Vistara,127859,NaN,NaN,NaN,NaN,NaN,NaN,NaN
flight,300153,1561,UK-706,3235,NaN,NaN,NaN,NaN,NaN,NaN,NaN
source_city,300153,6,Delhi,61343,NaN,NaN,NaN,NaN,NaN,NaN,NaN
departure_time,300153,6,Morning,71146,NaN,NaN,NaN,NaN,NaN,NaN,NaN
stops,300153,3,one,250863,NaN,NaN,NaN,NaN,NaN,NaN,NaN
arrival_time,300153,6,Night,91538,NaN,NaN,NaN,NaN,NaN,NaN,NaN
destination_city,300153,6,Mumbai,59097,NaN,NaN,NaN,NaN,NaN,NaN,NaN
class,300153,2,Economy,206666,NaN,NaN,NaN,NaN,NaN,NaN,NaN
duration,300153.0,NaN,NaN,NaN,12.221021,7.191997,0.83,6.83,11.25,16.17,49.83


## Missing Values + Dupicate Check

In [14]:
summary = pd.DataFrame({
    "Missing Values": df.isnull().sum(),
    "Missing %": (df.isnull().sum()/len(df))*100,
    "Unique Values": df.nunique(),
    "Data Types": df.dtypes
})

display(summary)

print("\nDuplicate Rows :", df.duplicated().sum())

,Missing Values,Missing %,Unique Values,Data Types
Unnamed: 0,0,0.0,300153,int64
airline,0,0.0,6,object
flight,0,0.0,1561,object
source_city,0,0.0,6,object
departure_time,0,0.0,6,object
stops,0,0.0,3,object
arrival_time,0,0.0,6,object
destination_city,0,0.0,6,object
class,0,0.0,2,object
duration,0,0.0,476,float64



Duplicate Rows : 0


## Target Variable Understanding

In [15]:
print(df["price"].describe())

print("\nMinimum Price :", df["price"].min())
print("Maximum Price :", df["price"].max())
print("Average Price :", round(df["price"].mean(),2))

count    300153.000000
mean      20889.660523
std       22697.767366
min        1105.000000
25%        4783.000000
50%        7425.000000
75%       42521.000000
max      123071.000000
Name: price, dtype: float64

Minimum Price : 1105
Maximum Price : 123071
Average Price : 20889.66


## Feature Understanding

In [16]:
feature_info = pd.DataFrame({
    "Feature": df.columns,
    "Data Type": df.dtypes.values,
    "Unique Values": [df[col].unique() for col in df.columns]
})

feature_info

,Feature,Data Type,Unique Values
0,Unnamed: 0,int64,"[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,..."
1,airline,object,"[SpiceJet, AirAsia, Vistara, GO_FIRST, Indigo,..."
2,flight,object,"[SG-8709, SG-8157, I5-764, UK-995, UK-963, UK-..."
3,source_city,object,"[Delhi, Mumbai, Bangalore, Kolkata, Hyderabad,..."
4,departure_time,object,"[Evening, Early_Morning, Morning, Afternoon, N..."
5,stops,object,"[zero, one, two_or_more]"
6,arrival_time,object,"[Night, Morning, Early_Morning, Afternoon, Eve..."
7,destination_city,object,"[Mumbai, Bangalore, Kolkata, Hyderabad, Chenna..."
8,class,object,"[Economy, Business]"
9,duration,float64,"[2.17, 2.33, 2.25, 2.08, 12.25, 16.33, 11.75, ..."


# Conclusion

✔ Dataset Loaded Successfully

✔ Total Rows : 300153

✔ Total Features : 12

✔ Target Variable : Price

✔ Numerical Features

✔ Categorical Features

✔ No Missing Values (verify after running)

✔ Ready for EDA